core/src/main/resources/hudson/model/CauseAction/summary.jelly

```diff
@@ -24,14 +24,32 @@ THE SOFTWARE.
 
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
-  <t:summary icon="orange-square.png">
-    <j:forEach var="entry" items="${it.causeCounts.entrySet()}">
-      <p>
-        <st:include page="description.jelly" it="${entry.key}"/>
-        <j:if test="${entry.value > 1}">
-          <st:nbsp/>${%Ntimes(entry.value)}
-        </j:if>
-      </p>
-    </j:forEach>
-  </t:summary>
+	<t:summary icon="orange-square.png">
+	<h4>${%Cause}</h4>
+	<style type="text/css">
+	div.tab:not(:target) div.content, div.tab:target div.show, div.tab div.hide {
+		display: none;
+	}
+	div.tab:target div.content,  div.tab:target div.hide {
+		display: block;
+	}
+	</style>
+	<div class="tab" id="text1">
+		<div class="show">		
+			<a href="#text1">${%Show}</a>
+			<br> </br>
+			</div>
+		<div class="hide">
+			<a href="#">${%Hide}</a>
+		</div>   
+		<div class="content">
```

Comment:
Oh I see - I think I mixed causes` up with upstream builds - what I actually wanted to say is, that the number of shown upstream builds should be limited and expandable. And then we could still show all causes for each of the builds.